src/wallet/wallet.cpp

```diff
@@ -2533,6 +2533,36 @@ void CWallet::ClearNoteWitnessCache()
     nWitnessCacheSize = 0;
 }
 
+template<typename NoteDataMap>
+void UpdateSpentHeightAndMaybePruneWitnesses(NoteDataMap& noteDataMap, int indexHeight, const uint256& nullifier)
```

Comment:
I disagree, because these internal functions are not complete state transitions that result in valid states. They are helpers to reduce code duplication in `{Increment, Decrement}NoteWitnesses` but don't make sense outside that context.

Reply:
We could however make these static (so they are even more clearly only usable in this file). I'll push a commit to do this.